tests/pmrfc3164-AtSignsInHostname.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# add 2016-11-22 by Pascal Withopf, released under ASL 2.0
+. $srcdir/diag.sh init
+. $srcdir/diag.sh generate-conf
+. $srcdir/diag.sh add-conf '
+module(load="../plugins/imtcp/.libs/imtcp")
+input(type="imtcp" port="13514" ruleset="customparser")
+parser(name="custom.rfc3164" type="pmrfc3164" permit.AtSignsInHostname="on")
+template(name="outfmt" type="string" string="-%hostname%-\n")
+
+ruleset(name="customparser" parser="custom.rfc3164") {
+	:syslogtag, contains, "tag" action(type="omfile" template="outfmt" file="rsyslog.out.log")
+}
+'
+. $srcdir/diag.sh startup
+. $srcdir/diag.sh tcpflood -m1 -M "\"<129>Mar 10 01:00:00 Hostname1 tag: msgnum:1\""
+. $srcdir/diag.sh tcpflood -m1 -M "\"<129>Mar 10 01:00:00 Hostn@me2 tag:  msgnum:2\""
+. $srcdir/diag.sh tcpflood -m1 -M "\"<129>Mar 10 01:00:00 Hostname3 tag:msgnum:3\""
+. $srcdir/diag.sh tcpflood -m1 -M "\"<129>Mar 10 01:00:00 Hos@name4 tag4:\""
+. $srcdir/diag.sh shutdown-when-empty
+. $srcdir/diag.sh wait-shutdown
```

Comment:
I think you should best also add a test for default behavior and the hostname containing an at sign.